packages/server/src/routes/agencies.js

```diff
@@ -16,20 +17,36 @@ const {
 
 router.get('/', requireUser, async (req, res) => {
     const { user } = req.session;
-    let response;
-    if (user.role.name === 'admin') {
-        response = await getAgencies(req.session.selectedAgency);
-    } else {
-        response = await getAgency(req.session.selectedAgency);
+    const response = await getTenantAgencies(user.tenant_id);
+    res.json(response);
+});
+
+router.get('/impersonable', requireAdminUser, async (req, res) => {
+    const { selectedAgency, user } = req.session;
+    const { asAgency } = req.params;
+    let agencyId = selectedAgency;
+
+    if (Number.isFinite(asAgency)) {
+        const allowed = await isUserAuthorized(user, asAgency);
+        if (!allowed) {
+            return res.status(403);
+        }
+        agencyId = asAgency;
     }
+    const response = await getAgencyTree(agencyId);
     res.json(response);
 });
 
 router.put('/:agency', requireAdminUser, async (req, res) => {
```

Comment:
nit: rename the path parameter to `:agencyId` to make its downstream use clearer. Same for the other routes in this file that have a `:agency` param.

Reply:
I'd rather not grow the scope of this PR with a bunch of consistency changes that are currently outside of the diff. I'll make another issue for it. The repo in general has a lot of inconsistencies. I've tried to just leave the code in one place consistent with itself.

Reply:
https://github.com/usdigitalresponse/usdr-gost/issues/328